src/common/config.h

```diff
@@ -40,6 +40,32 @@ enum {
 
 extern const char *ceph_conf_level_name(int level);
 
+namespace internal {
```

Comment:
Nit: since this only seems to apply to ```md_config_impl```, can we put this in a less-generic namespace than ```internal```?  If this is going to be re-used, perhaps it should be broken out into its own header.

Reply:
Understood re: private namespaces, my concern was that this is generically named in the root namespace (i.e. not under ```ceph```). I also thought that boost at least put its implementation namespace under a sub-namespace for the component it is related to (e.g. ```ceph::config::internal```).